solutions/usaco-815.mdx

```diff
@@ -0,0 +1,115 @@
+---
+id: usaco-815
+source: USACO Gold 2018 Febuary
+title: Taming the Herd
+author: Shijie Ren
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_gold_feb18.html)
+
+## Explanation
+
+Let $\texttt{dp}[i][j]$ be the the minimum number of changes that must be made
+to the first $j$ entries so that there are $i$ breakouts among the first $j$
+entries.
+
+There are three cases when we calculate the current value of $\texttt{dp}[i][j]$:
+
+- If the log was tampered with and the cows would NOT breakout:
+
+  - Compared to the previous day, the breakout number won't change. However, we need one
+    more change of log, so we transfer from $\texttt{dp}[i][j-1]$.
+    $$
+    \texttt{dp}[i][j]=\texttt{dp}[i][j-1]+1
+    $$
+
+- If log was tampered with and cows would breakout (Notice $a[j] \neq 0$):
+
+  - Compared to the previous day the breakout number increases by 1. We also
+    need one more change of the log, so we transfer from $\texttt{dp}[i-1][j-1]$.
+    $$
+    \texttt{dp}[i][j]=\texttt{dp}[i-1][j-1]+1
+    $$
+
+- If log was NOT tampered with:
+  - Then the $i$-th breakout would be on the $(j-a[j])$-th day, so we transfer from
+    $\texttt{dp}[i-1][j-a[j]-1]$.
+    $$
+    \texttt{dp}[i][j] = \texttt{dp}[i-1][j-a[j]-1]+\texttt{range\_ans}[j-a[j]][j]);
+    $$
+
+$$\texttt{range\_ans}[l][r]$$ represents the minimum cost to replace the range
+$a[l]...a[r]$ with $0...(r-l)$.
+
+Thus, our final DP relation is
+
+$$
+\texttt{dp}[i][j] = \max \{ \texttt{dp}[i][j-1]+1 ,\texttt{dp}[i-1][j-1]+1 ,\texttt{dp}[i-1][j-a[j]-1] + \texttt{range\_ans}[j-a[j]][j]) \}
```

Comment:
wait you said `max` but in the code you use `min`?

Reply:
oh no terrible mistake

Reply:
lol, dw we all do stuff like this